Dockerfile

```diff
@@ -1,11 +1,12 @@
-FROM node:14.21.3-buster-slim
+FROM node:16.20.2-alpine
 
 WORKDIR /usr/local
 
-COPY package-lock.json package-lock.json
+COPY yarn.lock yarn.lock
```

Comment:
Any specific reason to replace npm with yarn?

Reply:
For faster install speed :).

Reply:
pnpm is considered faster, more memory-efficient, and more performant than yarn, so why not use pnpm instead?

When trying to introduce/replace/ditch specific tooling, it's essential to 
- provide supplementary docs or materials to explain why this change is necessary.
- undergo impact analysis to understand the potential benefits and drawbacks.
- evaluate if there are other alternatives and make sure it's the right decision

To sum up, I believe using yarn instead would some what bring us some improvement, but it's not a good practice to just introduce it out of nowhere.

Reply:
> if we use npm install, it will encounter ERESOLVE overriding peer dependency

how and when this issue be introduced? it doesn't convince me that using yarn instead would just resolve the issue.

Reply:
pnpm sounds great. But for node version 16.
i don't think node version 16 has a great compatibility to it [ref](https://pnpm.io/installation#compatibility).
Maybe it's better for node version 18.

Reply:
fixed. Thanks

Reply:
it's still a good try though, we can start a discussion and have another PR for this 👍🏽